Source/Dafny/AST/DafnyAst.cs

```diff
@@ -392,6 +392,10 @@ public interface IBoundVarsBearingExpression : IRegion {
     }
   }
 
+  public interface IBoundVarBearingExpressionWithToken : IBoundVarsBearingExpression {
```

Comment:
What does this token mean? This interface already has a `BodyStartTok` and `BodyEndTok`. What's this third token?

Related but feel free to ignore: in the Roslyn compiler the type `SyntaxToken`, which every node in the AST is a subtype of, has a `Span` and `FullSpan` field, indicating the text it occupies without and with trivia (whitespace and comments).

Reply:
Good point. I needed to access only the bound variables and the token of an expression. Since after your remark I just figured out they all had `ComprehensionExpr` in common, I'm removing this interface.

Reply:
For the other point, it will be useful in my other work in progress about Trivia.